lms/djangoapps/courseware/user_state_client.py

```diff
@@ -53,12 +53,7 @@ def __init__(self, user=None):
         """
         self.user = user
 
-    @contract(
-        username="basestring",
-        block_key=UsageKey,
-        scope=ScopeBase,
-        fields="seq(basestring)|set(basestring)|None"
-    )
+    @donottrack(StudentModule, StudentModuleHistory)
```

Comment:
(and we use `ContractMeta`, which pushes the contracts from an abstract base class down into the derivative classes)